components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -1507,4 +1770,200 @@ private void setSPAttributeToRequest(HttpServletRequest req, String issuer, Stri
+     * @return Session Index.
+     * @throws IdentityException Decoding error.
+     */
+    private String extractSessionIndex(HttpServletRequest request) throws IdentityException {
+
+        String initialSamlLogoutRequest = request.getParameter(SAMLSSOConstants.SAML_REQUEST);
+        XMLObject samlRequest = SAMLSSOUtil.unmarshall(SAMLSSOUtil.decode(initialSamlLogoutRequest));
+
+        String sessionIndex = null;
+        if (samlRequest instanceof LogoutRequestImpl) {
+            sessionIndex = ((LogoutRequestImpl) samlRequest).getSessionIndexes().size() > 0 ?
+                    ((LogoutRequestImpl) samlRequest).getSessionIndexes().get(0).getSessionIndex() : null;
+        }
+
+        return sessionIndex;
+    }
+
+    /**
+     * Retrieves information of front-channel session participants in single logout.
+     *
+     * @param logoutRequestId Logout request id.
+     * @return Front-Channel SLO Participant Information.
+     */
+    private FrontChannelSLOParticipantInfo getFrontChannelSLOParticipantInfo(String logoutRequestId) {
+
+        FrontChannelSLOParticipantInfo frontChannelSLOParticipantInfo =
+                FrontChannelSLOParticipantStore.getInstance().getValueFromCache(logoutRequestId);
+
+        return frontChannelSLOParticipantInfo;
+    }
+
+    /**
+     * Removes information of front-channel slo session participant from the store.
+     *
+     * @param logoutRequestId Logout request id.
+     */
+    private void removeFrontChannelSLOParticipantInfo(String logoutRequestId) {
+
+        FrontChannelSLOParticipantStore.getInstance().clearCacheEntry(logoutRequestId);
+    }
+
+    /**
+     * This method is used to prepare a SAML request message as a HTTP query string for HTTP Redirect binding.
+     *
+     * @param logoutRequest     Logout Request.
+     * @param serviceProviderDO SAMLSSOServiceProviderDO.
+     * @return Redirect URL.
+     * @throws IdentityException Error in marshalling or setting signature to http query string.
+     */
+    private String createHttpQueryStringForRedirect(LogoutRequest logoutRequest,
+                                                    SAMLSSOServiceProviderDO serviceProviderDO)
+            throws IdentityException {
+
+        // Convert the SAML logout request object to a string.
+        String logoutRequestString = (SAMLSSOUtil.marshall(logoutRequest)).
+                replaceAll(SAMLSSOConstants.XML_TAG_REGEX, "").trim();
+
+        StringBuilder httpQueryString = null;
+        String signatureAlgorithmUri = serviceProviderDO.getSigningAlgorithmUri();
+
+        String tenantDomain = serviceProviderDO.getTenantDomain();
+        if (StringUtils.isEmpty(tenantDomain)) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+
+        try {
+            httpQueryString = new StringBuilder(SAMLSSOConstants.SAML_REQUEST + "=" +
+                    URLEncoder.encode(SAMLSSOUtil.compressResponse(logoutRequestString), "UTF-8"));
```

Comment:
Shall we use a constant for this "UTF-8" ?

Reply:
Fixed with 3bbc008dd576ca0169fde3b49fe4f505a99200ea